src/transformers/models/chameleon/processing_chameleon.py

```diff
@@ -0,0 +1,162 @@
+# coding=utf-8
+# Copyright 2024 Meta Inc. and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Processor class for Chameleon.
+"""
+
+from typing import List, Optional, Union
+
+from ...feature_extraction_utils import BatchFeature
+from ...image_utils import ImageInput
+from ...processing_utils import ProcessorMixin
+from ...tokenization_utils_base import PaddingStrategy, PreTokenizedInput, TextInput, TruncationStrategy
+from ...utils import TensorType
+
+
+class ChameleonProcessor(ProcessorMixin):
+    r"""
+    Constructs a Chameleon processor which wraps a Chameleon image processor and a Chameleon tokenizer into a single
+    processor.
+
+    [`ChameleonProcessor`] offers all the functionalities of [`ChameleonImageProcessor`] and [`LlamaTokenizerFast`].
+    See the [`~ChameleonProcessor.__call__`] and [`~ChameleonProcessor.decode`] for more information.
+
+    Args:
+        image_processor ([`ChameleonImageProcessor`]):
+            The image processor is a required input.
+        tokenizer ([`LlamaTokenizerFast`]):
+            The tokenizer is a required input.
+        image_seq_length (`int`, *optional*, defaults to 1024):
+            Sequence length of one image embedding.
+        image_token (`str`, *optional*, defaults to `"<image>"`):
+            The special token used to indicate image in the text.
+    """
+
+    attributes = ["image_processor", "tokenizer"]
+    tokenizer_class = ("LlamaTokenizer", "LlamaTokenizerFast")
+    image_processor_class = "ChameleonImageProcessor"
+
+    def __init__(self, image_processor, tokenizer, image_seq_length: int = 1024, image_token: str = "<image>"):
+        self.image_seq_length = image_seq_length
+        self.image_token = image_token
+        self.image_start_token = "<racm3:break>"  # fixed tokens for start and end, so can hardcode
+        self.image_end_token = "<eoss>"
```

Comment:
IMO we should still leave them as customizable!

Reply:
@ArthurZucker, @zucchini-nlp , we should probably move this to the tokenizer configs. That and the range of image tokens

Reply:
Sure, can make customizable also. About moving to the tokenizer, afaik we can't get `tokenizer.image_token` and extra added tokens will all be in `additional_special_tokens`, @ArthurZucker to confirm

If we can do that, happy to move to tokenizers :)

Reply:
I think it fits well in the processor TBH, but tokenizer not sure you can retain the info ok load / save given that you are using llama tokenizer. 
there are the CLS and SEP you can use ? 

Otherwise good time to think about some modularity for vLMs and these attributes!

Reply:
Right, nice to have some common special tokens (e.g. special_image_token) in tokenizer in the same way as we have BOS, EOS etc. I guess three more tokens should cover most cases: image, bei, eoi

Reply:
yeah I think generic properties / something more flexible than what we have right now would be nice